test/setups/extrude_geometry.jl

```diff
@@ -0,0 +1,136 @@
+            [0.0 0.0625 0.125 0.1875 0.25 0.3125 0.375 0.4375 0.5 0.0 0.0625 0.125 0.1875 0.25 0.3125 0.375 0.4375 0.5 0.0 0.0625 0.125 0.1875 0.25 0.3125 0.375 0.4375 0.5 0.0 0.0625 0.125 0.1875 0.25 0.3125 0.375 0.4375 0.5 0.0 0.0625 0.125 0.1875 0.25 0.3125 0.375 0.4375 0.5;
+             0.0 0.125 0.25 0.375 0.5 0.625 0.75 0.875 1.0 0.13975424859373686 0.26475424859373686 0.38975424859373686 0.5147542485937369 0.6397542485937369 0.7647542485937369 0.8897542485937369 1.0147542485937369 1.1397542485937369 0.2795084971874737 0.4045084971874737 0.5295084971874737 0.6545084971874737 0.7795084971874737 0.9045084971874737 1.0295084971874737 1.1545084971874737 1.2795084971874737 0.4192627457812106 0.5442627457812106 0.6692627457812106 0.7942627457812106 0.9192627457812106 1.0442627457812106 1.1692627457812106 1.2942627457812106 1.4192627457812106 0.5590169943749475 0.6840169943749475 0.8090169943749475 0.9340169943749475 1.0590169943749475 1.1840169943749475 1.3090169943749475 1.4340169943749475 1.5590169943749475],
+        ]
+
+        @testset "Direction $i" for i in eachindex(directions)
+            shape = ExtrudeGeometry((point1, point2); direction=directions[i], tlsph=true,
+                                    particle_spacing=0.15, n_extrude=5, density=1.0)
+
+            @test shape.coordinates ≈ expected_coords[i]
+        end
+    end
+
+    @testset verbose=true "Shifted Particles" begin
+        point1 = [0.0, 0.0]
+        point2 = [0.0, 1.0]
+
+        expected_coords = [-0.05 -0.05 -0.05 -0.05 -0.05 -0.05 -0.05 -0.05 -0.05 -0.05 -0.15 -0.15 -0.15 -0.15 -0.15 -0.15 -0.15 -0.15 -0.15 -0.15 -0.25 -0.25 -0.25 -0.25 -0.25 -0.25 -0.25 -0.25 -0.25 -0.25 -0.35 -0.35 -0.35 -0.35 -0.35 -0.35 -0.35 -0.35 -0.35 -0.35 -0.44999999999999996 -0.44999999999999996 -0.44999999999999996 -0.44999999999999996 -0.44999999999999996 -0.44999999999999996 -0.44999999999999996 -0.44999999999999996 -0.44999999999999996 -0.44999999999999996;
+                           0.05 0.15 0.24999999999999997 0.35 0.44999999999999996 0.55 0.65 0.7499999999999999 0.8499999999999999 0.95 0.05 0.15 0.24999999999999997 0.35 0.44999999999999996 0.55 0.65 0.7499999999999999 0.8499999999999999 0.95 0.05 0.15 0.24999999999999997 0.35 0.44999999999999996 0.55 0.65 0.7499999999999999 0.8499999999999999 0.95 0.05 0.15 0.24999999999999997 0.35 0.44999999999999996 0.55 0.65 0.7499999999999999 0.8499999999999999 0.95 0.05 0.15 0.24999999999999997 0.35 0.44999999999999996 0.55 0.65 0.7499999999999999 0.8499999999999999 0.95]
+        shape = ExtrudeGeometry((point1, point2); direction=[-1.0, 0.0],
+                                particle_spacing=0.1, n_extrude=5, density=1.0)
+
+        @test shape.coordinates ≈ expected_coords
+    end
+end
+
+# 3D
+@testset verbose=true "Extrude Geometry 3D" begin
```

Comment:
All of these lists are very long. Can you load them from csv?

Reply:
Hm, we also hard code the coordinates for the `RectangularTank` tests. Is it worth loading the coordinates from a separate file? It took me a while to generate this, and I need to generate this again for a csv file. Can we leave it as is and if it gets complicated we can still add a separate csv file. @efaulhaber what's your opinion on this?

On the other hand, it's a good point to put coordinates for setup tests in a separate csv file...

Reply:
Yes I also put long lists in the interpolation tests. But this is crap. Maybe we should agree to put these in csv files in the future?

Reply:
Let's discuss this in an issue.

Reply:
#490